Release 0.65.0 prep (#5548)
prometheus-operator/prometheus-operator: CHANGELOG.md

## 0.65.0 / 2022-05-04

The main change introduced by this release is the new v1alpha1 `ScrapeConfig` CRD.
This implements the [proposal](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/proposals/202212-scrape-config.md)
documented in [#5279](https://github.com/prometheus-operator/prometheus-operator/pull/5279)

> and provides a Kubernetes native API to create and manage additional scrape configurations.

Reviewer: adding a paragraph explaining how to enable the new CRD might be useful (similar to what we did in v0.64.0 for the PrometheusAgent CRD).

Author: Added, let me know if it looks good!

* [FEATURE] Add the `status` subresource for the `ThanosRuler` CRD. #5520
* [FEATURE] Add `spec.web.timeout` and `spec.web.getConcurrency` to the `Alertmanager` CRD. #5478
* [FEATURE] Add `spec.groups[].limit` to the `Prometheus` CRD. #4999
* [FEATURE] Add ScrapeConfig CRD. #5335
* [ENHANCEMENT] Set a default for `seccompProfile` on the operator and webhook Deployments to `RuntimeDefault`. #5477
* [ENHANCEMENT] Add optional liveness and readiness probes to prometheus-config-reloader. This can be enabled via the `--enable-config-reloader-probes` CLI flag. #5449

> * [ENHANCEMENT] Add HTTP `scheme` validation in `Probe.spec` and `PodMonitor.spec.podMetricsEndpoints`. #5531

Reviewer: I think that we can omit this PR because in the end, it updates only the description. We gave up enforcing the scheme value as it would have been a breaking change.

> * [BUGFIX] Add validation for missing CRDs. #5476

Reviewer: ```suggestion
* [BUGFIX] Don't start the PrometheusAgent controller if the CRD isn't present or the operator lacks permissions. #5476
```

## 0.65.0 / 2022-05-04

The main change introduced by this release is the new v1alpha1 `ScrapeConfig` CRD.
This implements the [proposal](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/proposals/202212-scrape-config.md)

> documented in [#5279](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/proposals/202212-scrape-config.md)

Reviewer 2: Is the intent to link to the issue rather than the proposal? The link is the same as the previous link.

and provides a Kubernetes native API to create and manage additional scrape configurations.

> * [CHANGE] Add validation for missing CRDs. #5476

Reviewer: it should probably be a BUGFIX since the original issue was that the `PrometheusOperatorNotReady` alert would fire if the PrometheusAgent CRD wasn't deployed. At least it's not a CHANGE.

* [FEATURE] Add the `status` subresource for the `ThanosRuler` CRD. #5520

> * [FEATURE] Add support for setting the `web.timeout` and `web.get-concurrency` options on `Alertmanager`

Reviewer: ```suggestion
* [FEATURE] Add `spec.web.timeout` and `spec.web.getConcurrency`  to the `Alertmanager` CRD. #5478
```

> * [FEATURE] Add support for `Prometheus` `RuleGroup` Limit. #4999

Reviewer: ```suggestion
* [FEATURE] Add `spec.groups[].limit` to the `Prometheus` CRD. #4999
```

* [FEATURE] Add ScrapeConfig CRD. #5335

> * [ENHANCEMENT] Set a default for `seccompProfile` on the operator Deployment to `RuntimeDefault`. #5477

Reviewer: ```suggestion
* [ENHANCEMENT] Set a default for `seccompProfile` on the operator and webhook Deployments to `RuntimeDefault`. #5477
```

> * [ENHANCEMENT] Add liveness and readiness probes to prometheus-config-reloader. #5449

Reviewer: ```suggestion
* [ENHANCEMENT] Add optional liveness and readiness probes to prometheus-config-reloader. This can be enabled via the `--enable-config-reloader-probes` CLI flag. #5449
```

* [ENHANCEMENT] Add HTTP `scheme` validation in `Probe.spec` and `PodMonitor.spec.podMetricsEndpoints`. #5531
* [BUGFIX] Declare `spec.rules` optional in `PrometheusRule` CRD. #5481
* [BUGFIX] Fix incorrect metric counter value for failed sync status. #5533

> * [BUGFIX] Report downstream errors when CRD validation fails. #5534

Reviewer: I wouldn't include it since IIUC the bug never made it in an official release.
